stub/rawkv.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stub
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// RawClientStub can be used like a rawkv.Client while it redirects all function
+// calls to an httpproxy server.
+type RawClientStub struct {
+	client      http.Client
+	proxyServer string
+	id          string
+}
+
+// NewRawClientStub creates a client for rawkv calls.
+func NewRawClientStub(proxyServer string, pdServers []string) (*RawClientStub, error) {
+	client := &RawClientStub{
+		client:      http.Client{Timeout: time.Second * 10},
+		proxyServer: strings.TrimSuffix(proxyServer, "/"),
+	}
+	res, err := client.send("/rawkv/client/new", &RawRequest{PDAddrs: pdServers})
+	if err != nil {
+		return nil, err
+	}
+	client.id = res.ID
+	return client, nil
+}
+
+// Close closes the client and releases resources in proxy server.
+func (c *RawClientStub) Close() error {
+	_, err := c.send(fmt.Sprintf("/rawkv/client/%s/close", c.id), &RawRequest{})
```

Comment:
maybe we can directly use `+` here?

Reply:
I think this way is more readable to understand the format of the URI.

Reply:
`fmt.Sprintf` relies on the reflection which is not elegant I think.

Reply:
Why reflection is not elegant... For the validator, performance is not important, what matters is correctness and readability.

Reply:
ok